### ISSUE TITLE
sql: add support for extract(string, interval)

### ISSUE THREAD

**Issue by kevinjqiu** [cockroachdb/cockroach#43209]
**Describe the problem**

Trying to use `EXTRACT(EPOCH FROM interval)`

**To Reproduce**

What did you do? Describe in your own words.

Running the following query on Cockroachdb v19.2.1

```
SELECT
    he1.ts,
    he2.query_type,
    he2.client_id,
    he1.host_name,
    EXTRACT(epoch FROM he2.ts - he1.ts) lat
FROM he AS he1 INNER JOIN he AS he2 ON he1.query_id = he2.query_id
WHERE he1.event_name = 'EVENT_1' AND he2.event_name = 'EVENT_2'
```

`he.ts` is a timestamp field.

**Expected behavior**
Expect the above query to return a result.

**Actual behaviour:**
Query failed with:

```
[42883] ERROR: unknown signature: extract(string, interval)
```

**Environment:**
 - CockroachDB: v19.2.1
 - Server OS: Container-Optimized OS (on GKE)
 - Client app: DataGrip (Java Postgres Client)

**Additional context**
What was the impact?

**Comment by otan** [comment]
looks like we have `extract_duration` but it is missing some units supported by postgres, see: `src/interfaces/ecpg/pgtypeslib/interval.c`

i'll give this a whirl.

**Comment by otan** [comment]
@awoods187 wdyt about supporting `extract(x from interval)`, and removing the `extract_duration` function? or should we leave extract_duration behind? if we leave it behind, should it do what it does now (only support a subset of extract arguments), or should it be an alias for `extract(x from interval)`?

**Comment by otan** [comment]
This will be fixed by #43293 and available in 20.1 (unfortunately, backporting to 19.2.3 is not tractable as extract returns the wrong type in 19.2, and this would introduce a weird inconsistency, and is thus a lot of effort).

You can currently work around this by casting the interval to a float/int/decimal. However, it differs from postgres behaviour for intervals >= 1 year -- returning 365 days of stuff instead of 365.25. This is separately tracked in #43272, but no guarantees it will be around in 20.1 ...
